File: core/src/ru/myitschool/dcrawler/utils/SeededRandom.java
package ru.myitschool.dcrawler.utils;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.utils.Logger;

import java.util.Random;

public class SeededRandom {

    private static long seed;
    private static final Random random;

    static {
        seed = new Random().nextLong();
        Gdx.app.debug("random seed", "" + seed);
        random = new Random(seed);
    }

    public static Random getInstance() {
        return getInstance(null);
    }

    public static Random getInstance(Long seed) {
        if (seed != null && !seed.equals(SeededRandom.seed)) {
            Gdx.app.debug("random seed", "" + seed);
            random.setSeed(seed);
        }
        return random;
    }

    public static long getSeed() {
        return seed;
    }
}

File: core/src/com/wolg_vlad/dcrawler/utils/SeededRandom.kt
package com.wolg_vlad.dcrawler.utils

import com.badlogic.gdx.Gdx
import java.util.*

object SeededRandom {
    var seed: Long = 0
    private var random: Random
    val instance: Random?
        get() = getInstance()

    fun getInstance(seed: Long = this.seed): Random {
        if (seed != SeededRandom.seed) {
            Gdx.app.debug("random seed", "" + seed)
            random!!.setSeed(seed)
        }
        return random
    }

    init {
        seed = Random().nextLong()
        Gdx.app.debug("random seed", "" + seed)
        random = Random(seed)
    }
}